src/Compilers/Core/MSBuildTask/InteractiveCompiler.cs

```diff
@@ -194,17 +216,15 @@ protected override int ExecuteTool(string pathToTool, string responseFileCommand
 
         protected override string GenerateCommandLineCommands()
         {
-            var commandLineBuilder = new CommandLineBuilderExtension();
-            AddCommandLineCommands(commandLineBuilder);
-            return commandLineBuilder.ToString();
+            return DotnetHostInfo.CommandLineArgs;
```

Comment:
As long as we don't change the arguments returned by the API (`Csc.CommandLineArgs`) I think that's good.